src/magnify/registry.py

```diff
@@ -29,11 +32,254 @@ def component_factory(*args, **kwargs):
+        The maximum number of iterations to perform during the button detection process, controlling the depth of search.
+    min_roundness :
+        The minimum roundness value for detected buttons. Buttons that do not meet this roundness threshold are excluded.
+    cluster_penalty :
+        The penalty factor for clustering during button detection. This parameter helps control how strongly nearby buttons are grouped together.
+    roi_length :
+        The length (in pixels) of the region of interest (ROI) around detected buttons.
+    progress_bar :
+        If True, display a progress bar during processing to track the progress of the pipeline.
+    search_timestep :
+        A list of time steps to focus on for button detection. If provided, only the specified time steps are used. If `None`, all time steps are used.
+    search_channel :
+        The channel or list of channels to use for button detection and expression filtering. If `None`, all channels will be used. Channels are used to compute foreground and background values for contrast-based filtering.
+    squeeze :
+        If True, removes singleton dimensions from the dataset(dimensions of size 1).
+    roi_only :
+        If True, only returns the region of interest(roi) from the dataset, ignoring other parts of the image.
+    drop_tiles :
+        If True, removes the "tile" variable from the dataset after stitching, effectively dropping individual tile data.
+
+    Returns
+    -------
+    pipe : Pipeline
+        The constructed pipeline object, either configured and returned if `return_pipe` is True, or executed as part of the image processing workflow.
+
+    Notes
+    -----
+    This function constructs a pipeline architecture to process image data. The steps in the pipeline include:
+
+    - 'identify_buttons' : Identifies buttons based on the provided `pinlist` or `shape` parameters and assigns valid markers.
+    - 'stitch' : Stitches image tiles based on the overlap parameter.
+    - 'rotate' : Rotates the stitched image by the specified angle.
+    - 'find_buttons' : Detects buttons based on edge detection and clustering.
+    - 'filter_expression' : Filters buttons based on foreground-background contrast differences using a minimum contrast threshold.
+    - 'filter_nonround' : Filters out buttons that do not meet the roundness criteria.
+    - 'filter_leaky' : Removes buttons that are determined to be leaky or poorly segmented.
+    - 'drop' : Optionally removes unnecessary tiles and simplifies the dataset based on the `squeeze`, `roi_only`, and `drop_tiles` options.
+
+    Examples
+    --------
+    >>> mini_chip_pipe = mini_chip(channels=[0], pinlist='pinlist.csv', overlap=100, rotation=45)
+    >>> mini_chip = mini_chip_pipe(data=my_image_data)
+
+    This processes `my_image_data` by stitching tiles with 100 pixels of overlap, rotating the image by 45 degrees, using channel 0 for detection, and utilizing a pin layout from 'pinlist.csv'.
+    """
```

Comment:
So you don't have to keep updating two very big docstrings that are going to be almost identical you could just remove Notes, Returns, and Parameters and instead have a reference to the mini_chip function in the top-level docstring description. i.e.
Build the pipeline to find buttons in minichip images. See :func:`.mini_chip` for further details on usage.

Reply:
Resolved